From f8d7cf0ba61520167c66c93f444f29160449d13d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 4 Jun 2020 15:02:27 -0400
Subject: [PATCH] dhcp6: Force DUID-LLT
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

LXD VMs set the SMBIOS UUID but guest operating systems usually do not
use it for DHCPv6 DUID. As we have users who need to be able to use
MAC-based config in their DHCP servers, we'll just force the use of
DUID-LLT instead.

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 NetworkPkg/Dhcp6Dxe/Dhcp6Utility.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/NetworkPkg/Dhcp6Dxe/Dhcp6Utility.c b/NetworkPkg/Dhcp6Dxe/Dhcp6Utility.c
index 9eb1bdd..5352193 100644
--- a/NetworkPkg/Dhcp6Dxe/Dhcp6Utility.c
+++ b/NetworkPkg/Dhcp6Dxe/Dhcp6Utility.c
@@ -66,7 +66,7 @@ Dhcp6GenerateClientId (
   //
   // If System UUID is found from SMBIOS Table, use DUID-UUID type.
   //
-  if ((PcdGet8 (PcdDhcp6UidType) == Dhcp6DuidTypeUuid) && !EFI_ERROR (NetLibGetSystemGuid (&Uuid)) && !CompareGuid (&Uuid, &gZeroGuid)) {
+  if (0 && (PcdGet8 (PcdDhcp6UidType) == Dhcp6DuidTypeUuid) && !EFI_ERROR (NetLibGetSystemGuid (&Uuid)) && !CompareGuid (&Uuid, &gZeroGuid)) {
     //
     //
     //  The format of DUID-UUID:
